clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -446,6 +448,9 @@ public void removeSensor(String name) {
                             removeMetric(metric.metricName());
                         log.debug("Removed sensor with name {}", name);
                         childSensors = childrenSensors.remove(sensor);
+                        for (final Sensor parent : sensor.parents()) {
+                            childrenSensors.getOrDefault(parent, emptyList()).remove(sensor);
```

Comment:
It's really weird that we store a reference to the parent sensors inside the `Sensor` object, but children sensors are kept in a separate map. 😕

Reply:
I frowned upon it as well, and for the history the parent stored on sensor was there since day one, while the children map is added a year later. I think it's actually better to keep both the parents and children inside a single Sensor, but we can do that in another cleanup PR.

Reply:
Yeah, I fought the urge to refactor it into a more traditional bidirectionally linked graph structure.

Maybe later; it doesn't seem worth opening that can of worms right now.